trace_replay/trace_replay.cc

```diff
@@ -329,26 +328,28 @@ Status Replayer::MultiThreadReplay(uint32_t threads_num) {
         replay_epoch + std::chrono::microseconds(
                            (ra->trace_entry.ts - header.ts) / fast_forward_));
     if (ra->trace_entry.type == kTraceWrite) {
-      thread_pool.Schedule(&Replayer::BGWorkWriteBatch, ra, nullptr, nullptr);
+      thread_pool.Schedule(&Replayer::BGWorkWriteBatch, ra.release(), nullptr,
+                           nullptr);
       ops++;
     } else if (ra->trace_entry.type == kTraceGet) {
-      thread_pool.Schedule(&Replayer::BGWorkGet, ra, nullptr, nullptr);
+      thread_pool.Schedule(&Replayer::BGWorkGet, ra.release(), nullptr,
+                           nullptr);
       ops++;
     } else if (ra->trace_entry.type == kTraceIteratorSeek) {
-      thread_pool.Schedule(&Replayer::BGWorkIterSeek, ra, nullptr, nullptr);
+      thread_pool.Schedule(&Replayer::BGWorkIterSeek, ra.release(), nullptr,
+                           nullptr);
       ops++;
     } else if (ra->trace_entry.type == kTraceIteratorSeekForPrev) {
-      thread_pool.Schedule(&Replayer::BGWorkIterSeekForPrev, ra, nullptr,
-                           nullptr);
+      thread_pool.Schedule(&Replayer::BGWorkIterSeekForPrev, ra.release(),
+                           nullptr, nullptr);
       ops++;
-    } else {
-      // It is the case that ra->trace_entry.type == kTraceEnd
-      assert(ra->trace_entry.type == kTraceEnd);
-
+    } else if (ra->trace_entry.type == kTraceEnd) {
       // Do nothing for now, free the ReplayerWorkerArg.
       // TODO: Add some validations later.
-      delete ra;
       break;
+    } else {
+      // Other trace entry types that are not replayed.
```

Comment:
Wait, I thought we should never reach here?

Reply:
No....I have tested in one real trace, it reach there. There might be some other trace entry types that are tracked but not replayed. I'm thinking not to assert it but do continue.

Reply:
@riversand963 Actually, it is safe to use continue. For example, some one implemented more tracing types in the tracer, but replay has not been developed yet. It will switch to "else", we should not stop replay. If the trace is broken and there is no KTraceEnd in the entry, ReadTrace the Status will be "Corrupted" and break out.

Reply:
Thanks for the clarification. If the program can reach this point, then we should not assert(false).